ldes-server-domain/src/main/java/be/vlaanderen/informatievlaanderen/ldes/server/domain/ldesfragment/services/FragmentationServiceImpl.java

```diff
@@ -6,28 +6,38 @@
 import be.vlaanderen.informatievlaanderen.ldes.server.domain.ldesfragment.entities.LdesFragment;
 import be.vlaanderen.informatievlaanderen.ldes.server.domain.ldesfragment.repository.LdesFragmentRepository;
 import be.vlaanderen.informatievlaanderen.ldes.server.domain.ldesfragment.valueobjects.FragmentInfo;
+import be.vlaanderen.informatievlaanderen.ldes.server.domain.ldesfragmentrequest.valueobjects.LdesFragmentRequest;
 import be.vlaanderen.informatievlaanderen.ldes.server.domain.ldesmember.entities.LdesMember;
 import be.vlaanderen.informatievlaanderen.ldes.server.domain.ldesmember.repository.LdesMemberRepository;
 
 import java.util.List;
+import java.util.Optional;
 
 public class FragmentationServiceImpl implements FragmentationService {
 	private final LdesFragmentRepository ldesFragmentRepository;
 	private final LdesMemberRepository ldesMemberRepository;
 	private final LdesConfig ldesConfig;
 
 	public FragmentationServiceImpl(LdesFragmentRepository ldesFragmentRepository,
-			LdesMemberRepository ldesMemberRepository, LdesConfig ldesConfig) {
+			LdesMemberRepository ldesMemberRepository, LdesConfig ldesConfig, String name) {
 		this.ldesFragmentRepository = ldesFragmentRepository;
 		this.ldesMemberRepository = ldesMemberRepository;
 		this.ldesConfig = ldesConfig;
-		addRootFragment();
+		addRootFragment(name);
 	}
 
-	private void addRootFragment() {
+	private void addRootFragment(String viewName) {
+		Optional<LdesFragment> optionalRoot = ldesFragmentRepository
+				.retrieveFragment(new LdesFragmentRequest(ldesConfig.getCollectionName(), viewName, List.of()));
+		if (optionalRoot.isEmpty()) {
+			createRoot(viewName);
+		}
```

Comment:
Since, this method doesn't return anything (void interface), I'd prefer it to look as is. If it would return a rootFragment, I could agree that `retrieveFragment().orElseGet(()->createRoot())` would be okay. Now such a statement will return an LdesFragment, but we don't do anything with it.